test/config-luatest/vshard_test.lua

```diff
@@ -0,0 +1,365 @@
+local treegen = require('test.treegen')
+local server = require('test.luatest_helpers.server')
+local helpers = require('test.config-luatest.helpers')
+
+local g = helpers.group()
+
+local has_vshard = pcall(require, 'vshard')
+
+g.test_fixed_masters = function(g)
+    t.skip_if(not has_vshard, 'Module "vshard" is not available')
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = [[
+    credentials:
+      users:
+        guest:
+          roles: [super]
+
+    iproto:
+      listen: 'unix/:./{{ instance_name }}.iproto'
+      advertise:
+        sharding: 'storage:storage@'
+
+    sharding:
+      bucket_count: 1234
+      sched_ref_quota: 258
+
+    groups:
+      group-001:
+        replicasets:
+          replicaset-001:
+            database:
+              replicaset_uuid: '11111111-1111-1111-0011-111111111111'
+            instances:
+              instance-001:
+                database:
+                  mode: rw
+                sharding:
+                  roles: [storage]
+              instance-002:
+                sharding:
+                  roles: [storage]
+          replicaset-002:
+            instances:
+              instance-003:
+                database:
+                  instance_uuid: '22222222-2222-2222-0022-222222222222'
+                  mode: rw
+                sharding:
+                  roles: [storage]
+              instance-004:
+                sharding:
+                  roles: [storage]
+          replicaset-003:
+            instances:
+              instance-005:
+                database:
+                  mode: rw
+                sharding:
+                  roles: [router]
+    ]]
+    local config_file = treegen.write_script(dir, 'config.yaml', config)
+    local opts = {
+        env = {LUA_PATH = os.environ()['LUA_PATH']},
+        config_file = config_file,
+        chdir = dir,
+    }
+    g.server_1 = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+    g.server_2 = server:new(fun.chain(opts, {alias = 'instance-002'}):tomap())
+    g.server_3 = server:new(fun.chain(opts, {alias = 'instance-003'}):tomap())
+    g.server_4 = server:new(fun.chain(opts, {alias = 'instance-004'}):tomap())
+    g.server_5 = server:new(fun.chain(opts, {alias = 'instance-005'}):tomap())
+
+    g.server_1:start({wait_until_ready = false})
+    g.server_2:start({wait_until_ready = false})
+    g.server_3:start({wait_until_ready = false})
+    g.server_4:start({wait_until_ready = false})
+    g.server_5:start({wait_until_ready = false})
+
+    g.server_1:wait_until_ready()
+    g.server_2:wait_until_ready()
+    g.server_3:wait_until_ready()
+    g.server_4:wait_until_ready()
+    g.server_5:wait_until_ready()
+
+    -- Check that cluster was created.
+    local info = g.server_1:eval('return box.info')
+    t.assert_equals(info.name, 'instance-001')
+    t.assert_equals(info.replicaset.name, 'replicaset-001')
+    t.assert_equals(info.cluster.name, 'group-001')
```

Comment:
It'll fail after rebase, see 07e6060bcfde9d379342d50acf9ea40722437ee4.

Reply:
Thanks, fixed.